client/hostvolumemanager/host_volumes.go

```diff
@@ -220,3 +247,29 @@ func genVolConfig(req *cstructs.ClientHostVolumeCreateRequest, resp *HostVolumeP
 		ReadOnly: false,
 	}
 }
+
+// nameLocker is used to ensure that volumes on each node are unique by name.
+// The volume scheduler will prevent this too, but only after node fingerprint,
+// so we need to protect against concurrent duplicate creates.
+type nameLocker struct {
+	locks sync.Map
+}
+
+// lock the provided name, error if it was already locked
+func (l *nameLocker) lock(name, id string) error {
+	currentID, exists := l.locks.LoadOrStore(name, id)
+	if exists && id != currentID.(string) {
+		return fmt.Errorf("%w: name=%q id=%q", ErrVolumeNameExists, name, id)
```

Comment:
The caller has the name and ID and already logs, so we should probably return only `ErrVolumeNameExists` here and then have the caller log the error rather than formatting this by hand and having the formatting mismatch for JSON loggers.